extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/StackTracePropagator.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.extensions.trace.propagation;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A propagator designed to inject and extract multiple trace {@code HttpTextFormat} propagators,
+ * intendended for backwards compatibility with existing services using different formats. It works
+ * in a stack-fashion, starting with the last registered propagator, to the first one.
```

Comment:
Agree with this. I don't see any reason to go in a stack-like order; it'll be much clearer is just to go in order, like @anuraaga suggests.

Reply:
This is based on prior art (also, see #1273 ) but I'm free to change that if we all agree that we want simple linear usage.

cc @pavolloffay

Reply:
+1 to simple linear order

Reply:
One observation is this is about how to set propagation order for a Java developer. It seems like this doesn't really need a spec but could be left up to the language? FIFO, LIFO, it may be idiomatic one way or another in different languages. In Java, I think going in-order is idiomatic, we have some famous examples like servlet filter chain. But I can also imagine a function language may use something like b3(w3c) which could be an idiomatic way of expressing the w3c, then b3, order in that language. Environment variables, being a non-programming concept, do make more sense for a spec though.

Just a thought.

Reply:
Actually I'd like this to be in the spec, but it might take a little bit of time before we can have it there (*if* we actually manage to have it there).

I'm not familiar with servlet filter, but is it something that traverses *all* components, or can it stop right away upon a successful operation? If that's latter, that can help us go forward with this approach (always, we can go linear and change it later on).

For the sake of this PR, I will change to name to `TraceMultiPropagator` now, while we wait for feedback.